src/bit-fields-macros/src/builder.rs

```diff
@@ -0,0 +1,816 @@
+                        } else {
+                            Err(serde::de::Error::custom("no 2nd value in seq"))
+                        }
+                    } else {
+                        Err(serde::de::Error::custom("no 1st value in seq"))
+                    }
+                }
+            }
+            impl<'de> serde::Deserialize<'de> for #struct_name {
+                fn deserialize<D>(deserializer: D) -> Result<#struct_name, D::Error>
+                where
+                    D: serde::Deserializer<'de>,
+                {
+                    deserializer.deserialize_tuple(2,#visitor)
+                }
+            }
+        };
+        #[cfg(not(feature = "serde"))]
+        let serde = TokenStream::new();
+
+        #[cfg(feature = "set_theory")]
+        let set_theory = quote!(
+            /// Returns `true` if `self` is a [`superset`](https://en.wikipedia.org/wiki/Subset) of `other`.
+            pub fn superset(&self, other: &Self) -> bool {
+                #fields_superset_fn
+            }
+            /// Returns `true` if `self` is a [`subset`](https://en.wikipedia.org/wiki/Subset) of `other`.
+            pub fn subset(&self, other: &Self) -> bool {
+                #fields_subset_fn
+            }
+            /// Returns `true` if `self` and `other` are [`disjoint sets`](https://en.wikipedia.org/wiki/Disjoint_sets).
+            pub fn disjoint(&self, other: &Self) -> bool {
+                #fields_disjoint_fn
+            }
+            /// Returns the [`intersection`](https://en.wikipedia.org/wiki/Intersection_(set_theory)) of `self` and `other`.
+            pub fn intersection(&self, other: &Self) -> Self {
+                let mut base = Self::from(0);
+                #fields_intersection_fn
+                base
+            }
+            /// Returns the [`union`](https://en.wikipedia.org/wiki/Union_(set_theory)) of `self` and `other`.
+            pub fn union(&self, other: &Self) -> Self {
+                let mut base = Self::from(0);
+                #fields_union_fn
+                base
+            }
+            /// Returns [`std::cmp::Ordering`] based on bit flags.
+            /// - `Some(Ordering::Equal)` - Bit flags match between `self` and `other`.
+            /// - `Some(Ordering::Greater)` - Bit flags of `self` are a strict superset of bit flags of `other`.
+            /// - `Some(Ordering::Less)` - Bit flags of `self` are a strict subset of bit flags of `other`.
+            /// - `None` - None of the above conditions are met.
+            pub fn cmp_flags(&self,other: &Self) -> Option<std::cmp::Ordering> {
+                if self.equal(other) {
+                    Some(std::cmp::Ordering::Equal)
+                }
+                else if self.superset(other) {
+                    Some(std::cmp::Ordering::Greater)
+                }
+                else {
+                    Some(std::cmp::Ordering::Less)
+                }
+            }
```

Comment:
This never returns `None`. Instead it returns `Ordering::Less` if the two bitstrings are not comparable.
```suggestion
            /// Returns [`std::cmp::Ordering`] based on bit flags.
            /// - `Some(Ordering::Equal)` - Bit flags match between `self` and `other`.
            /// - `Some(Ordering::Greater)` - Bit flags of `self` are a strict superset of bit flags of `other`.
            /// - `Some(Ordering::Less)` - Bit flags of `self` are a strict subset of bit flags of `other`.
            /// - `None` - None of the above conditions are met.
            pub fn cmp_flags(&self,other: &Self) -> Option<std::cmp::Ordering> {
                if self.equal(other) {
                    Some(std::cmp::Ordering::Equal)
                } else if self.superset(other) {
                    Some(std::cmp::Ordering::Greater)
                } else  if other.superset(self) {
                    Some(std::cmp::Ordering::Less)
                } else  {
                    None
                }
            }
```

Reply:
I remember some issues here, but what you suggest does seem right I will implement and do some testing.